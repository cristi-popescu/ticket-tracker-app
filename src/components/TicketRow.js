import React from "react";

const TicketRow = props => {
    const formatDate = date => {
        const options = {
            year: "numeric",
            month: "short",
            day: "numeric",
            hour: "numeric",
            hour12: false,
            minute: "numeric",
            second: "numeric"
        };
        const formattedDate = new Date(date);

        if (!isNaN(formattedDate.getTime())) {
            return formattedDate.toLocaleDateString("en-EN", options);
        } else {
            return "N/A";
        }
    };

    const displayValue = (key, value) => {
        const dateColumns = ["creationDate", "lastModified"];

        return dateColumns.indexOf(key) === -1 ? value : formatDate(value);
    };

    const { ticket } = props;
    const { tableColumns, sortingKey } = props;
    const tableCells = Object.keys(tableColumns).map(key => {
        return (
            <td key={key} className={sortingKey === key ? "sorted" : ""}>
                {displayValue(key, ticket[key])}
            </td>
        );
    });

    return <tr>{tableCells}</tr>;
};

export default TicketRow;
