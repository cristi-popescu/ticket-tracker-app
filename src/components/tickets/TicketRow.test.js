import React from "react";
import TicketRow from "./TicketRow";
import renderer from "react-test-renderer";

describe("TicketRow", () => {
  test("renders correctly", () => {
    const ticketRowProps = {
      ticket: {
        status: "1",
        severity: "3",
        name: "Ticket 1",
        description: "Lorem ipsum 1",
        lastModified: "2019-11-09T15:56:11.623Z",
        key: "TCK-1",
        creationDate: "2019-10-07T20:40:53.003Z",
        id: "kXsAeFG"
      },
      tableColumns: {
        key: "Key",
        name: "Name",
        status: "Status",
        creationDate: "Creation Date",
        lastModified: "Last Modified"
      },
      sortingRule: {
        sortingKey: "key",
        direction: "asc"
      }
    };
    const component = renderer.create(<TicketRow {...ticketRowProps} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
