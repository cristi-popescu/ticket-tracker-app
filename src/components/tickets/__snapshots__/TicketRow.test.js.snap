// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TicketRow renders correctly 1`] = `
<tr
  onClick={[Function]}
>
  <td
    className=""
  >
    <b>
      TCK-1
    </b>
  </td>
  <td
    className=""
  >
    Ticket 1
  </td>
  <td
    className=""
  >
    <span
      className="badge"
    >
      1
    </span>
  </td>
  <td
    className=""
  >
    Oct 7, 2019 11:40 PM
  </td>
  <td
    className=""
  >
    Nov 9, 2019 5:56 PM
  </td>
</tr>
`;
